tests/Unit/RmStepTest.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+use Symfony\Component\Filesystem\Filesystem;
+use WordPress\Blueprints\BlueprintException;
+use WordPress\Blueprints\Model\DataClass\RmStep;
+use WordPress\Blueprints\Runner\Step\RmStepRunner;
+use WordPress\Blueprints\Runtime\NativePHPRuntime;
+
+
+beforeEach(function() {
+    $this->runtime = new NativePHPRuntime(sys_get_temp_dir());
+
+    $this->step = new RmStepRunner();
+    $this->step->setRuntime($this->runtime);
+
+    $this->fileSystem = new Filesystem();
+});
+
+it('should remove a directory when an absolute path is provided', function() {
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+    $this->fileSystem->mkdir($resolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $resolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it('should remove a directory when a relative path is provided', function () {
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+    $this->fileSystem->mkdir($resolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $unresolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it ('should remove a directory with a subdirectory', function () {
+    $unresolvedSubDir = "dir/subdir";
+    $resolvedSubDir = $this->runtime->resolvePath($unresolvedSubDir);
+    $this->fileSystem->mkdir($resolvedSubDir);
+
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $unresolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it ('should remove a directory with a file', function () {
+    $unresolvedFile = "dir/file.txt";
+    $resolvedFile = $this->runtime->resolvePath($unresolvedFile);
+    $this->fileSystem->dumpFile($resolvedFile, "test");
+
+    $unresolvedDirToRemove = "dir";
+    $resolvedDirToRemove = $this->runtime->resolvePath($unresolvedDirToRemove);
+
+    $input = new RmStep();
+    $input->path = $unresolvedDirToRemove;
+
+    $this->step->run($input);
+
+    expect($this->fileSystem->exists($resolvedDirToRemove))->toBeFalse();
+});
+
+it ('should remove a file', function () {
+   $unresolvedFileToRemove = "file.txt";
+   $resolvedFileToRemove = $this->runtime->resolvePath($unresolvedFileToRemove);
+   $this->fileSystem->dumpFile($resolvedFileToRemove, "test");
+
+   $input = new RmStep();
+   $input->path = $unresolvedFileToRemove;
+
+   $this->step->run($input);
+
+   expect($this->fileSystem->exists($resolvedFileToRemove))->toBeFalse();
+});
+
+it ('should throw an exception when asked with a relative path to remove a nonexistent directory ', function() {
```

Comment:
```suggestion
it ('should throw an exception when asked to remove a nonexistent directory (using a relative path) ', function() {
```

Reply:
Let me know if you agree, but I'd stick with `nonexistent`. This is a nitpick, really. Historically hyphens were used in BrE, and a single word in AmE. Generally, the trend is to accept nonhyphenated words in BrE, too. Could we stick to the AmE spelling, as we use behavior instead of behaviour.

The other thing - sure. Will do.